tools/train.py

```diff
@@ -53,6 +53,11 @@ def parse_args():
         default='none',
         help='job launcher')
     parser.add_argument('--local_rank', type=int, default=0)
+    parser.add_argument(
+        '--runner-type',
+        choices=['epoch', 'iter'],
+        default='iter',
+        help='Type of mmcv runner to use. EpochBased or IterBased.')
```

Comment:
In the current implementation, we use args to determine which runner to use. 
However, some config fields are also related to the runner type. We may try to choose runner type from config file. So that users could directly use EpochRunner by changing config. 
For example, if there is no `runner_type` in the config, `IterRunner` will be used. Otherwise, `EpochRunner` will be used.

Reply:
Following that path, maybe it's a good idea to add a "build_runner" using Registry to `mmcv`, so the configuration is similar to the "optimizer" or other fields that use the registry?? 

It would also allow the users to easily try custom runners

Reply:
Yep. Sounds good. And we could set the default runner to be `IterRunner` for MMSeg. 
You may create a PR in MMCV to add a runner registry.

Reply:
Will do. Making this a draft untill mmcv is updated